File: chapitre-03/exemple-execvp.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-execvp.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------

from __future__ import print_function
import os

args = [ "ls", "-l", "-n" ]

try:
	os.execvp(args[0], args)
except OSError:
	print("Error, unable to run", args[0])

File: chapitre-03/exemple-execvp.c
// ------------------------------------------------------------------
// exemple-execvp.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <errno.h>

int main (void)
{
	char * argv[] = { "ls", "-l", "-n", (char *) NULL };

	execvp("ls", argv);

	fprintf(stderr, "Error %d\n", errno);
	return 1;
}

